comp/snmptraps/listener/component.go

```diff
@@ -0,0 +1,24 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+// Package listener implements a component that listens for SNMP messages,
+// parses them, and publishes messages on a channel.
+package listener
+
+import (
+	"github.com/DataDog/datadog-agent/comp/snmptraps/packet"
+)
+
+// team: network-device-monitoring
+
+// Component is the component type.
+type Component interface {
+	// Start starts the listener
+	Start() error
```

Comment:
Can you use [Lifecycle hooks](https://uber-go.github.io/fx/lifecycle.html) instead of `Start` method?

Reply:
Done.